docs/howto/composite-primary-key.txt

```diff
@@ -0,0 +1,114 @@
+=====================================
+How to define a composite primary key
+=====================================
+
+In Django, each model has a primary key.
+By default, this primary key consists of a single field.
+
+In most cases, a single-field primary key should suffice.
+In database design, however, defining a primary key that consists of multiple
+fields is sometimes necessary.
+
+To define a composite primary key, add the ``models.CompositePrimaryKey``
+field to your model:
+
+.. code-block:: pycon
+
+    class Tenant(models.Model):
+        pass
+
+    class User(models.Model):
+        pk = models.CompositePrimaryKey("tenant_id", "id")
+        tenant = models.ForeignKey(Tenant, on_delete=models.CASCADE)
+        id = models.IntegerField()
+
+This will instruct Django to create a composite ``PRIMARY KEY`` constraint
+instead of a single-field primary key.
+
+It's possible to define a composite primary key consisting of a single field.
+This is practically the same as setting ``primary_key=True`` on the field.
+
+.. code-block:: pycon
+
+    class Model(models.Model):
+        pk = models.CompositePrimaryKey("id")
+        id = models.IntegerField()
+
+Migrating to a composite primary key
+====================================
+
+As of now, Django doesn't offer built-in support for the migration of primary
+keys.
+
+This means once a table is created, its primary key cannot be changed via
+Django's migration operations.
```

Comment:
Sure, we can document it!

Unfortunately, changing a table's primary key is a complex and hairy procedure.
Django doesn't really support changing the pk via migrations, composite pk is no exception.

Reply:
One thing we might also be able to do here is provide a more readable error when running `makemigrations`. Do we have anything like this for normal pk changes?

Reply:
There's no special error, it usually just fails with a db error.

Reply:
So this case didn't fail when applying the migration and perhaps it might be nice to have alterfield on CompositePrimaryKeys to fail loudly

> maybe we should (or already do) have a ticket for this in the normal pk case?

Agree a ticket would be great, we should be specific on what this is. 
We do have some tickets that are in this [rough "area"](https://code.djangoproject.com/query?component=Migrations&description=~primary+key&status=assigned&status=new&order=id&desc=1&col=id&col=summary&col=status&col=component&col=owner&col=type&col=version).

Reply:
Afaik, it's the same "no operation" behaviour as running an AlterField with `primary_key=False` or `primary_key=True`.
So if we want an error, it should also take that into consideration .
I'm all for making this as user friendly as possible, I could add this but maybe in another PR. I can create a ticket for it tomorrow.

Reply:
I adjusted the docs.

Reply:
https://code.djangoproject.com/ticket/35589#ticket